apps/notion-api/src/app/api/users/cron/run-sync-user/service.ts

```diff
@@ -0,0 +1,129 @@
+import { PrismaClient } from '@prisma/client';
+import axios from 'axios';
+
+const prisma = new PrismaClient();
+
+type PersonType = {
+    email : string
+}
+
+type UserResponseType = {
+    id: string;
+    name : string;
+    person: PersonType;
+}
+
+type UsersResponseType = {
+    results: UserResponseType[];
+    next_cursor: string;
+    has_more: boolean;
+}
+
+type UserData = {
+    id: string;
+    displayName: string;
+    email: string;
+}
+
+export const runUsersSyncJob = async () => {
+    try {
+        const job = await prisma.users_Sync_Jobs.findFirst({
+            orderBy: [
+                { priority: 'asc' }, // Order by priority in ascending order
+                { sync_started_at: 'asc' }, // Order by timestamp in ascending order
+            ],
+        });
```

Comment:
This code add condition on `retry_after`. Because it seems like (I am not sure) we can't use postgresql "now()" directly with prisma and `new Date()` isn't safe enough to use, I have to do a raw query here.
```suggestion
        const job: SomeType = await prisma.$queryRaw`SELECT * FROM "users_Sync_Jobs" WHERE "retry_after" < now() ORDER BY "priority" ASC, "sync_started_at" ASC LIMIT 1`;
```